File: regex-test/java/TesterAsync.java
import java.util.Scanner;
import java.util.regex.Pattern;
import java.util.regex.PatternSyntaxException;

public class TesterAsync {
    public static void main(String[] args) {
        try (Scanner input = new Scanner(System.in)) {
            while (input.hasNext()) {
                String regex = input.nextLine();
                if (!regex.startsWith("REGEX:")) {
                    continue;
                }

                try {
                    Pattern p = Pattern.compile("(?U)" + regex.substring(6));
                    System.out.printf("success\n");

                    while (input.hasNext()) {
                        String line = input.nextLine();
                        if (line.startsWith("TEST:")) {
                            String test = line.substring(5);
                            if (p.matcher(test).matches()) {
                                System.out.printf("test good\n");
                            } else {
                                System.out.printf("Regex '%s' does not match '%s'\n", regex, test);
                                break;
                            }
                        } else {
                            break;
                        }
                    }
                } catch (PatternSyntaxException e) {
                    String message = e.getMessage()
                            .replaceAll("\\\\", "\\\\\\\\")
                            .replaceAll("\n", "\\\\n");
                    System.out.printf("%s\n", message);
                }
            }
        }
    }
}

File: regex-test/python/tester_async.py
import sys
import re
import fileinput

def substituteLf(s):
    return s.replace("\\", "\\\\").replace("\n", "\\n")

regex = None

try:
    for line in fileinput.input():
        if line.endswith("\r\n"):
            line = line[:-2]
        elif line.endswith("\n"):
            line = line[:-1]

        if regex == None:
            try:
                regex = re.compile(line)
                print("success")
            except Exception as e:
                print(substituteLf(str(e)))
            sys.stdout.flush()
        elif line.startswith("TEST:"):
            test = line[5:]
            if regex.match(test) != None:
                print("test good")
            else:
                print(substituteLf("Regex '" + regex.pattern + "' does not match '" + test + "'"))
                regex = None
            sys.stdout.flush()
        else:
            regex = None
except KeyboardInterrupt:
    pass
